compiler-core/src/ast/constant.rs

```diff
@@ -71,6 +73,29 @@ impl TypedConstant {
             | Constant::Var { typ, .. } => typ.clone(),
         }
     }
+
+    pub fn private_fn_deps(&self) -> Vec<EcoString> {
```

Comment:
I may be confused here, but I don't think I'm traversing the constants twice! The bit in `erlang.rs` is traversing module definitions and the function in `constant.rs` is traversing the args for each constant!

Reply:
This one hasn't been done yet btw!

Reply:
Sorry, I think I misunderstood the issue!
Just to make it clear, you mean this part, right?
```
 TypedConstant::Record { args, .. } => args
                .iter()
                .for_each(|arg| arg.value.private_fn_deps(already_found)),
```

Reply:
Sorry, I'm reading the code and trying to understand the problem but I don't see how I'm traversing the constants a second time. The `module.definitions.iter()` in `erlang.rs` is traversing the constants, but the code quoted above traverses the deps of each constant. I don't see any duplication of efforts. Sorry if I'm being dumb here, but I really don't see the problem!

Reply:
The `private_fn_deps` function traverses the AST. Rather than have a function that does this let's keep track of any private usage while type checking the AST.

Reply:
I think it's actually fixed this time!